clients/osv/src/main/kotlin/OsvService.kt

```diff
@@ -92,15 +92,12 @@ interface OsvService {
 }
 
 @Serializable
-class VulnerabilitiesForPackageRequest private constructor(
+class VulnerabilitiesForPackageRequest(
     val commit: String? = null,
     @SerialName("package")
     val pkg: Package? = null,
     val version: String? = null
-) {
-    constructor(commit: String, pkg: Package? = null) : this(commit = commit, pkg = pkg, version = null)
-    constructor(pkg: Package, version: String) : this(commit = null, pkg = pkg, version = version)
-}
+)
```

Comment:
Why do these constructors get dropped?

Reply:
Because I would have needed to add a third one that allows to construct a `VulnerabilitiesForPackageRequest` from only a `Package` with a `purl`, but felt that this was too much overhead, so went the opposite direction and deleted all of them to give the caller full freedom over which properties to specify.

And actually these constructors add less safety than it seems at first, because even with the property-combinations implied by the constructor you could get things wrong. For example, using `constructor(pkg: Package, version: String)` would fail if the `Package` contains a `purl` due to the duplicate `version` properties then.